### DISCUSSION TITLE
cluster with etcd, get error: The specified bucket does not exist

### DISCUSSION THREAD

**Discussion by luokeqiang** [minio/minio#20714] [Q&A] [answered]
I deployed a cluster with etcd。
I created a bucket, but I cannot operate it and get the error message "The specified bucket does not exist"。
I don't know what's wrong with me?

### minio conf
- minio5: 192.168.10.135
- minio6: 192.168.10.136
```
# /etc/default/minio

MINIO_ETCD_ENDPOINTS="http://192.168.10.128:2379,http://192.168.10.129:2379,http://192.168.10.130:2379"
MINIO_PUBLIC_IPS=192.168.10.135,192.168.10.136
MINIO_DOMAIN=minio.domain2.com

MINIO_VOLUMES="http://minio{5...6}/mnt/data{1...2}/minio"

MINIO_OPTS="--console-address :9001"
MINIO_ROOT_USER=minio
MINIO_ROOT_PASSWORD=minio123
```
### etcd conf
```
#[Member]
ETCD_DATA_DIR="/data/etcd"
ETCD_LISTEN_PEER_URLS="http://0.0.0.0:2380"
ETCD_LISTEN_CLIENT_URLS="http://0.0.0.0:2379"
ETCD_NAME="etcd1"

#[Clustering]
ETCD_INITIAL_ADVERTISE_PEER_URLS="http://192.168.10.128:2380"
ETCD_ADVERTISE_CLIENT_URLS="http://192.168.10.128:2379"
#ETCD_INITIAL_CLUSTER="etcd1=http://192.168.10.128:2380"
ETCD_INITIAL_CLUSTER="etcd1=http://192.168.10.128:2380,etcd2=http://192.168.10.129:2380,etcd3=http://192.168.10.130:2380"
ETCD_INITIAL_CLUSTER_TOKEN="etcd-cluster"
ETCD_INITIAL_CLUSTER_STATE="new"

```

![1732965388029](https://github.com/user-attachments/assets/5d93059c-ff18-40b8-af16-b3b8020662c7)

![1732966553483](https://github.com/user-attachments/assets/9055d5b1-8709-40f7-93e5-de849f8480ec)

etcd data
![1732966370938](https://github.com/user-attachments/assets/20c1b2cc-adbd-4a7f-a5db-566048d8631a)

## Minio service
```
[Unit]
Description=MinIO
Documentation=https://min.io/docs/minio/linux/index.html
Wants=network-online.target
After=network-online.target
AssertFileIsExecutable=/usr/local/bin/minio

[Service]
WorkingDirectory=/usr/local

User=minio-user
Group=minio-user
#ProtectProc=invisible

EnvironmentFile=-/etc/default/minio
ExecStartPre=/bin/bash -c "if [ -z \"${MINIO_VOLUMES}\" ]; then echo \"Variable MINIO_VOLUMES not set in /etc/default/minio\"; exit 1; fi"
ExecStart=/usr/local/bin/minio server $MINIO_OPTS $MINIO_VOLUMES

# MinIO RELEASE.2023-05-04T21-44-30Z adds support for Type=notify (https://www.freedesktop.org/software/systemd/man/systemd.service.html#Type=)
# This may improve systemctl setups where other services use `After=minio.server`
# Uncomment the line to enable the functionality
# Type=notify

# Let systemd restart this service always
Restart=always

# Specifies the maximum file descriptor number that can be opened by this process
LimitNOFILE=65536

# Specifies the maximum number of threads this process can create
TasksMax=infinity

# Disable timeout logic and wait until process is stopped
TimeoutStopSec=infinity
SendSIGKILL=no

[Install]
WantedBy=multi-user.target

# Built for ${project.name}-${project.version} (${project.name})

```

### minio version
```
minio version RELEASE.2024-11-07T00-52-20Z (commit-id=cefc43e4daa4cbb490ef6726ea374e26a93eb85e)
Runtime: go1.23.3 linux/amd64
License: GNU AGPLv3 - https://www.gnu.org/licenses/agpl-3.0.html
Copyright: 2015-2024 MinIO, Inc.
```
### etcd version
```
etcd Version: 3.5.17
Git SHA: 507c0de
Go Version: go1.22.9
Go OS/Arch: linux/amd64
```
### Linux
```
Distributor ID:	Ubuntu
Description:	Ubuntu 24.10
Release:	24.10
Codename:	oracular

Linux minio5 6.11.0-8-generic #8-Ubuntu SMP PREEMPT_DYNAMIC Mon Sep 16 13:41:20 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux

```

**Comment by klauspost** [comment, marked as answer]
Check the server logs. 

Why use etcd in the first place? In 99% af of all cases it is just an overcomplication.

**Reply by luokeqiang** [reply]
I'm just learning to use the etcd cluster solution first.

 I used `mc admin trace -a -v minio5`  and only got the following logs. I didn't see any other error logs.

![image](https://github.com/user-attachments/assets/1ae90359-007f-4c68-b09f-571eeadfaba4)

![image](https://github.com/user-attachments/assets/742b86ee-dded-40fe-b1ff-3f30dfdcd60f)

**Reply by harshavardhana** [reply]
etcd bucket federation is deprecated, please use etcd without bucket federation. 

> MINIO_DOMAIN=minio.domain2.com

without this..

**Reply by luokeqiang** [reply]
@harshavardhana I once removed this configuration, but without it, couldn't send bucket registration information to etcd. 
Etcd only contains login credentials, I'm puzzled by this.
Are there any other properties that need to be configured?
![image](https://github.com/user-attachments/assets/bd62b058-a343-45e6-a13c-17f7a4b97c69)

**Reply by luokeqiang** [reply]
My apologies. I see now that etcd federation scaling is deprecated. What are the current alternative solutions?

**Reply by bennybi** [reply]
Why etcd federation scaling is deprecated, in which where to say it?

**Reply by wangxinyii** [reply]
> etcd bucket federation is deprecated, please use etcd without bucket federation.
> 
> > MINIO_DOMAIN=minio.domain2.com
> 
> without this..

What method can I use to dynamically expand capacity

**Comment by wangxinyii** [comment]
What method can I use to dynamically expand capacity

**Reply by marktheunissen** [reply]
https://min.io/docs/minio/linux/operations/install-deploy-manage/expand-minio-deployment.html
